modules/lvgl/lvgl.c

```diff
@@ -21,58 +22,86 @@
+		 IS_EQ(CONFIG_LV_Z_MONOCHROME_CONVERSION_BUFFER, 1))
 
 #ifdef CONFIG_LV_Z_BUFFER_ALLOC_STATIC
 
-#define DISPLAY_WIDTH  DT_PROP(DISPLAY_NODE, width)
-#define DISPLAY_HEIGHT DT_PROP(DISPLAY_NODE, height)
+#define DISPLAY_WIDTH(n)  DT_PROP(DISPLAY_NODE(n), width)
+#define DISPLAY_HEIGHT(n) DT_PROP(DISPLAY_NODE(n), height)
 
 #if IS_MONOCHROME_DISPLAY
 /* monochrome buffers are expected to have 8 preceding bytes for the color palette */
-#define BUFFER_SIZE                                                                                \
-	(((CONFIG_LV_Z_VDB_SIZE * ROUND_UP(DISPLAY_WIDTH, 8) * ROUND_UP(DISPLAY_HEIGHT, 8)) /      \
+#define BUFFER_SIZE(n)                                                                             \
+	(((CONFIG_LV_Z_VDB_SIZE * ROUND_UP(DISPLAY_WIDTH(n), 8) *                                  \
+	   ROUND_UP(DISPLAY_HEIGHT(n), 8)) /                                                       \
 	  100) / 8 +                                                                               \
 	 8)
 #else
-#define BUFFER_SIZE                                                                                \
+#define BUFFER_SIZE(n)                                                                             \
 	(CONFIG_LV_Z_BITS_PER_PIXEL *                                                              \
-	 ((CONFIG_LV_Z_VDB_SIZE * DISPLAY_WIDTH * DISPLAY_HEIGHT) / 100) / 8)
+	 ((CONFIG_LV_Z_VDB_SIZE * DISPLAY_WIDTH(n) * DISPLAY_HEIGHT(n)) / 100) / 8)
 #endif /* IS_MONOCHROME_DISPLAY */
 
-/* NOTE: depending on chosen color depth buffer may be accessed using uint8_t *,
- * uint16_t * or uint32_t *, therefore buffer needs to be aligned accordingly to
- * prevent unaligned memory accesses.
- */
-static uint8_t buf0[BUFFER_SIZE]
-#ifdef CONFIG_LV_Z_VDB_CUSTOM_SECTION
-	Z_GENERIC_SECTION(.lvgl_buf)
-#endif
-		__aligned(CONFIG_LV_Z_VDB_ALIGN);
+static uint32_t disp_buf_size[DT_ZEPHYR_DISPLAYS_COUNT] = {0};
+static uint8_t *buf0_p[DT_ZEPHYR_DISPLAYS_COUNT] = {NULL};
 
 #ifdef CONFIG_LV_Z_DOUBLE_VDB
-static uint8_t buf1[BUFFER_SIZE]
-#ifdef CONFIG_LV_Z_VDB_CUSTOM_SECTION
-	Z_GENERIC_SECTION(.lvgl_buf)
+static uint8_t *buf1_p[DT_ZEPHYR_DISPLAYS_COUNT] = {NULL};
 #endif
-		__aligned(CONFIG_LV_Z_VDB_ALIGN);
-#endif /* CONFIG_LV_Z_DOUBLE_VDB */
 
 #if ALLOC_MONOCHROME_CONV_BUFFER
-static uint8_t mono_vtile_buf[BUFFER_SIZE]
-#ifdef CONFIG_LV_Z_VDB_CUSTOM_SECTION
-	Z_GENERIC_SECTION(.lvgl_buf)
+static uint8_t *mono_vtile_buf_p[DT_ZEPHYR_DISPLAYS_COUNT] = {NULL};
 #endif
-		__aligned(CONFIG_LV_Z_VDB_ALIGN);
-#endif /* ALLOC_MONOCHROME_CONV_BUFFER */
+
+/* NOTE: depending on chosen color depth, buffers may be accessed using uint8_t *,*/
+/* uint16_t * or uint32_t *, therefore buffer needs to be aligned accordingly to */
+/* prevent unaligned memory accesses. */
+
+/* clang-format off */
+#define LV_DISPLAY_MACRO(n)                                                                        \
```

Comment:
Nit naming, this relates to the display buffers, so should be part of it I guess.